CardParts/src/Classes/CardPartsBottomSheetViewController.swift

```diff
@@ -0,0 +1,525 @@
+    public func presentBottomSheet(on hostView: UIView? = UIApplication.shared.keyWindow) {
+        guard let hostView = hostView, !isShowingBottomSheet else { return }
+        self.view.translatesAutoresizingMaskIntoConstraints = false
+        
+        hostView.addSubview(self.view)
+        
+        NSLayoutConstraint.activate([
+            self.view.leadingAnchor.constraint(equalTo: hostView.leadingAnchor),
+            self.view.trailingAnchor.constraint(equalTo: hostView.trailingAnchor),
+            self.view.bottomAnchor.constraint(equalTo: hostView.bottomAnchor)
+        ])
+        
+        setup()
+        
+        viewTopConstraint = self.view.topAnchor.constraint(equalTo: hostView.topAnchor)
+        if !shouldIncludeOverlay {
+            viewTopConstraint = self.view.topAnchor.constraint(equalTo: hostView.bottomAnchor, constant: -bottomSheetHeight)
+        }
+        viewTopConstraint.isActive = true
+
+        self.view.layoutIfNeeded()
+        
+        // animate appearing on screen
+        UIView.animate(withDuration: appearAnimationDuration, delay: 0, usingSpringWithDamping: 1.0, initialSpringVelocity: 1.0, options: UIView.AnimationOptions.curveEaseIn, animations: {
+            self.darkOverlay.alpha = self.overlayMaxAlpha
+            self.bottomSheetTopConstraint.constant -= self.bottomSheetHeight
+            self.view.layoutIfNeeded()
+        }, completion: { finished in
+            self.didShow?()
+            self.isShowingBottomSheet = true
+        })
+    }
+    
+    /// Adds a shadow to the bottom sheet container with the specified properties
+    public func addShadow(color: CGColor = UIColor.black.cgColor, opacity: Float = 0.2, radius: CGFloat = 4, offset: CGSize = CGSize(width: 0, height: 1)) {
+        bottomSheetContainerVC.view.layer.shadowColor = color
+        bottomSheetContainerVC.view.layer.shadowRadius = radius
+        bottomSheetContainerVC.view.layer.shadowOffset = offset
+        bottomSheetContainerVC.view.layer.shadowOpacity = opacity
+    }
+    
+    /// Updates the private height properties and constraints, and animates the bottom sheet to new height if it is currently showing.
+    public func updateHeight() {
+        calculateHeight()
+        
+        guard isShowingBottomSheet else { return }
+        // guard this because when reusing bottom sheet to avoid crash when sometimes constraints are mysteriously disappearing
+        guard self.bottomSheetTopConstraint != nil, self.viewTopConstraint != nil else {
+            self.dismissBottomSheet(.programmatic(info: ["error": "closing due to nil constraints"]))
+            return
+        }
+        UIView.animate(withDuration: changeHeightAnimationDuration, delay: 0, usingSpringWithDamping: 1.0, initialSpringVelocity: 1.0, options: UIView.AnimationOptions.curveEaseIn, animations: {
+            self.darkOverlay.alpha = self.overlayMaxAlpha
```

Comment:
Can we allow user to animation type which they need ?

Reply:
Good idea, added a variable for this too 👍